### PR TITLE
Use chatId from URL rather than from payload for chats

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
The verify access to a chat, we use HandleRequest() with the chatId provided. Currently, we get this from the payload, which can differ from the chatId from the URL, which opens us to a security problem where a user could inject an arbitrary chatId in the payload, which doesn't match what's in the URL.

### Description
- Use chatId from URL and only from URL
- Add integrations test to validate this

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: integration-tests/ChatTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using CopilotChat.WebApi.Models.Response;
+using Xunit;
+using static CopilotChat.WebApi.Models.Storage.CopilotChatMessage;
+
+namespace ChatCopilotIntegrationTests;
+
+public class ChatTests : ChatCopilotIntegrationTest
+{
+    [Fact]
+    public async void ChatWithBot()
```
Comment (alliscode) [inline integration-tests/ChatTests.cs:17]: Can you make this name more targeted? Maybe something like `ChatMessagePostSucceedsWithValidInput`.

**Review by alliscode — APPROVED** [review]

File: webapi/Controllers/ChatController.cs [hunk 2]
```diff
@@ -415,6 +412,28 @@ private async Task RegisterPlannerHostedFunctionsUsedAsync(CopilotChatPlanner pl
         return;
     }
 
+    private static ContextVariables GetContextVariables(Ask ask, IAuthInfo authInfo, string chatId)
```
[the remark this answers is not in the document]
Comment (glahaye) [reply webapi/Controllers/ChatController.cs:415]: It was to simplify, yes.

File: webapi/Controllers/ChatController.cs [hunk 3]
```diff
@@ -415,6 +412,28 @@ private async Task RegisterPlannerHostedFunctionsUsedAsync(CopilotChatPlanner pl
...
+    {
+        const string UserIdKey = "userId";
+        const string UserNameKey = "userName";
+        const string ChatIdKey = "chatId";
+
+        var contextVariables = new ContextVariables(ask.Input);
+        foreach (var variable in ask.Variables)
+        {
+            if (variable.Key != UserIdKey && variable.Key != UserNameKey)
```
Comment (TaoChenOSU) [inline webapi/Controllers/ChatController.cs:424]: I know this is copied from somewhere else. But what is the point of this condition? We are going to override the values anyways.

Comment (glahaye) [reply webapi/Controllers/ChatController.cs:424]: Yes, this check is redundant.  I'll remove it.

**Review by TaoChenOSU — APPROVED** [review]

**Pull request merged**